src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -10,21 +10,34 @@
 using Microsoft.ApplicationInsights;
 using Microsoft.ApplicationInsights.DataContracts;
 using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.Functions.Worker.ApplicationInsights.Initializers;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.Functions.Worker.ApplicationInsights
 {
     internal sealed class FunctionsTelemetryModule : ITelemetryModule, IAsyncDisposable
     {
         private const string DependencyTelemetryKey = "_tel";
         private const string DependencyTypeInProc = "InProc";
+        private readonly FunctionsApplicationInsightsOptions _options;
 
         private TelemetryClient? _telemetryClient;
         private ActivityListener? _listener;
 
+        public FunctionsTelemetryModule(IOptions<FunctionsApplicationInsightsOptions> options)
+        {
+            _options = options.Value;
+        }
+
         public void Initialize(TelemetryConfiguration configuration)
         {
             _telemetryClient = new TelemetryClient(configuration);
 
+            if (_options.TokenCredentialOptions != null)
+            {
+                configuration.SetAzureTokenCredential(_options.TokenCredentialOptions.CreateTokenCredential());
```

Comment:
I don't think setting it here is correct. I can't remember exactly but I believe this may be a copy of `TelemetryConfiguration`.

Instead, I think you need to do the following as part of options configuration of `TelemetryConfiguration`

Reply:
Done.